src/java/org/apache/cassandra/index/sai/utils/PrimaryKey.java

```diff
@@ -74,66 +100,103 @@ public PrimaryKey createPartitionKeyOnly(DecoratedKey partitionKey)
-                // to maintain the correct clustering sort order
-                ByteSource clusteringComparable = clustering() == null ||
-                                                  clustering().isEmpty() ? null
-                                                                         : clusteringComparator.asByteComparable(clustering())
-                                                                                               .asComparableBytes(version);
-                return ByteSource.withTerminator(version == ByteComparable.Version.LEGACY ? ByteSource.END_OF_STREAM
-                                                                                          : ByteSource.TERMINATOR,
-                                                 keyComparable,
-                                                 clusteringComparable);
+                return Kind.Token;
             }
 
             @Override
-            @SuppressWarnings("ConstantConditions")
-            public int compareTo(PrimaryKey o)
+            public Token token()
             {
-                int cmp = token().compareTo(o.token());
+                return token;
+            }
 
-                // If the tokens don't match then we don't need to compare any more of the key.
-                // Otherwise, if either of the keys are token only we can only compare tokens
-                if ((cmp != 0) || isTokenOnly() || o.isTokenOnly())
-                    return cmp;
+            @Override
+            public DecoratedKey partitionKey()
+            {
+                throw new UnsupportedOperationException();
+            }
 
-                // Next compare the partition keys. If they are not equal or
-                // this is a single row partition key or there are no
-                // clusterings then we can return the result of this without
-                // needing to compare the clusterings
-                cmp = partitionKey().compareTo(o.partitionKey());
-                if (cmp != 0 || hasEmptyClustering() || o.hasEmptyClustering())
-                    return cmp;
-                return clusteringComparator.compare(clustering(), o.clustering());
+            @Override
+            public Clustering<?> clustering()
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public ByteSource asComparableBytes(Version version)
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public int compareTo(PrimaryKey o)
+            {
+                return token().compareTo(o.token());
             }
 
             @Override
             public int hashCode()
             {
-                return Objects.hash(token(), partitionKey(), clustering(), clusteringComparator);
+                return Objects.hash(token(), 16, 32, clusteringComparator);
```

Comment:
What do the 16 and 32 represent?

Reply:
Ah, I think that got left in. I added them to try and avoid hash collisions but I'm not sure that is possible here or even needed. AFAICT token only keys are only used for comparison. 

I have removed these and replaced the magic number in the `SkinnyPrimaryKey` with `Clustering.EMPTY`.